setmatcher/src/main/java/dk/dbc/rr/oai/setmatcher/Worker.java

```diff
@@ -129,6 +134,31 @@ public boolean isInBadState() {
         return inBadState;
     }
 
+    public boolean queueStalled() {
+        int queueStalledAfter = config.getQueueStalledAfter();
+        if (System.currentTimeMillis() - lastDequeue.get() < queueStalledAfter * 1_000L) {
+            return false; // dequeued within the last 10 min.
+        }
+        try (Connection connection = rawRepo.getConnection() ;
+             PreparedStatement stmt = connection.prepareStatement("SELECT EXTRACT('EPOCH' FROM NOW() - queued)::INT FROM queue WHERE worker = 'oai-set-matcher' ORDER BY queued LIMIT 1")) {
```

Comment:
Kender ikke lige til sql kode, men tager den højde for sommer/vinter-tidsskift...?

Reply:
Sålænge det er "timestamp with time zone", når man så laver aritmetikken i postgresql instansen, så burde det virke, og så mange lægger ikke på kø i den time hver halve år... Så det kan godt være den ruller rundt et par gange der, hvis der bliver kørt et eller andet op der.